cognite/client/data_classes/contextualization.py

```diff
@@ -1036,3 +1038,213 @@ def __init__(
+
+    Example:
+
+        Configure a call to digrams detect endpoint:
+
+            >>> from cognite.client import CogniteClient
+            >>> from cognite.client.data_classes.contextualization import ConnectionFlags, DiagramDetectConfig
+            >>> client = CogniteClient()
+            >>> config = DiagramDetectConfig(
+            ...     remove_leading_zeros=True,
+            ...     connection_flags=ConnectionFlags(
+            ...         no_text_inbetween=True,
+            ...         natural_reading_order=True,
+            ...     )
+            ... )
+            >>> job = client.diagrams.detect(entities=[{"name": "A1"}], file_id=123, config=config)
+
+        If you want to use an undocumented parameter, you can pass it as a keyword argument:
+
+            >>> config_with_undoducmented_params = DiagramDetectConfig(
+            ...     remove_leading_zeros=True,
+            ...     connection_flags=ConnectionFlags(new_undocumented_flag=True),
+            ...     new_undocumented_param={"can_be_anything": True},
+            ... )
+    """
+
+    def __init__(
+        self,
+        annotation_extract: bool | None = None,
+        case_sensitive: bool | None = None,
+        connection_flags: ConnectionFlags | list[str] | None = None,
+        customize_fuzziness: CustomizeFuzziness | dict[str, Any] | None = None,
+        direction_delta: float | None = None,
+        direction_weights: DirectionWeights | dict[str, Any] | None = None,
+        min_fuzzy_score: float | None = None,
+        read_embedded_text: bool | None = None,
+        remove_leading_zeros: bool | None = None,
+        substitutions: dict[str, list[str]] | None = None,
+        **params: Any,
+    ) -> None:
+        self.annotation_extract = annotation_extract
+        self.case_sensitive = case_sensitive
+        self.connection_flags = connection_flags
+        self.customize_fuzziness = customize_fuzziness
+        self.direction_delta = direction_delta
+        self.direction_weights = direction_weights
+        self.min_fuzzy_score = min_fuzzy_score
+        self.read_embedded_text = read_embedded_text
+        self.remove_leading_zeros = remove_leading_zeros
+        self.substitutions = substitutions
+        self._extra_params = {}
+
+        _known_params = {to_camel_case(k): k for k in vars(self)}
+        for param_name, value in params.items():
+            if known := _known_params.get(to_camel_case(param_name)):
+                raise ValueError(
+                    f"Provided parameter name `{param_name}` collides with a known parameter `{known}`. Please use it insead."
+                )
+            else:
+                self._extra_params[param_name] = value
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        dumped = super().dump(camel_case=camel_case)
+
+        for param_name, v in self._extra_params.items():
+            dumped[to_camel_case(param_name) if camel_case else param_name] = v
```

Comment:
I don't understand what you mean. I'm just respecting the `camel_case` param. It's true by default anyways. If for some reason the user wants to dump the object in snake case I think it's more weird that some of the config options are converted and some are not.

I'm thinking that the user can stick to the pythonic snake_case when using the config class but could also use camelCase, so we don't really know which case the use will provide when initializing the class.

When loading we expect the values to come from the api, in camelCase, since it's not arbitrary metadata but part of the api schema, albeit not documented.

Reply:
All I'm trying to say is that a roundtrip using `to_camel_case` and `to_snake_case` can only be considered a no-op for well-behaved strings, that's why I think you should not pass them through either of these functions, not in load and not in dump.

If a user is passing these super-custom settings, they need to account for CorrectCasing themselves. Just my 2 cents (I'll leave the decision up to you!)

Reply:
(A bit like how we in data modeling aren't case converting properties)